### PR TITLE
fix(minion): fix requisites from recent prs

### LINKED ISSUE #475: [BUG] Recent MacOS commits cause problems in managing salt services
### Formula commit hash / release tag
master

### Versions reports (master & minion)
Salt Version:
           Salt: 3000.3
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.8
         Jinja2: 2.10
        libgit2: Not Installed
       M2Crypto: 0.27.0
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.17 (default, Apr 15 2020, 17:20:14)
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 16.0.2
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.5
 
System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-101-generic
         system: Linux
        version: Ubuntu 18.04 bionic

### Pillar / config used
I'm using saltclass to manage my salt minions, and I am also using gitfs to obtain my salt formulas. Here are the basic pillars I'm using that should be relevant for testing this:

```
pillars:
  salt:
    minion:
      acceptance_wait_time: 10
      acceptance_wait_time_max: 120
      auth_timeout: 180
      id: "{{__grains__['id']}}"
      keysize: 2048
      master: "192.168.1.217"
      master_tries: "-1"
      max_event_size: 100000000
      random_reauth_delay: 270
      recon_default: 1000
      recon_max: 60000
      recon_randomize: True
      use_superseded:
        - ^
        - module.run
```

---

### Describe the bug
When attempting to run the salt.minion state or the salt.master state, I receive the following set of errors:

```
  Name: salt-minion - Function: pkg.installed - Result: Clean Started: - 06:57:29.191378 Duration: 1511.518 ms
  Name: /etc/salt/minion.d - Function: file.recurse - Result: Clean Started: - 06:57:30.703287 Duration: 923.891 ms
----------
          ID: salt-minion
    Function: service.running
      Result: False
     Comment: The following requisites were not found:
                                 watch:
                                     file: salt-master-macos
                                     file: salt-master
     Started: 06:57:31.634685
    Duration: 0.08 ms
     Changes:   
  Name: /etc/salt/minion.d/_defaults.conf - Function: file.absent - Result: Clean Started: - 06:57:31.644005 Duration: 0.893 ms
  Name: salt-call --local service.restart salt-minion --out-file /dev/null - Function: cmd.run - Result: Clean Started: - 06:57:31.646149 Duration: 0.015 ms
```

```
----------
          ID: salt-master
    Function: pkg.installed
      Result: True
     Comment: All specified packages are already installed
     Started: 07:05:19.978614
    Duration: 32.142 ms
     Changes:   
----------
          ID: salt-master
    Function: file.recurse
        Name: /etc/salt/master.d
      Result: True
     Comment: The directory /etc/salt/master.d is in the correct state
     Started: 07:05:20.013480
    Duration: 343.485 ms
     Changes:   
----------
          ID: salt-master
    Function: service.running
      Result: False
     Comment: The following requisites were not found:
                                 watch:
                                     file: salt-master-macos
     Started: 07:05:20.357944
    Duration: 0.008 ms
     Changes:   
----------
          ID: remove-old-master-conf-file
    Function: file.absent
        Name: /etc/salt/master.d/_defaults.conf
      Result: True
     Comment: File /etc/salt/master.d/_defaults.conf is not present
     Started: 07:05:20.357996
    Duration: 0.283 ms
     Changes:   
```

This appears to be a new condition, as I did not receive these errors yesterday.

### Steps to reproduce the bug
1) Put pillars in place.
2) Attempt to run the salt.minion state on a minion.
3) See error.

### Expected behaviour
I should not receive any errors when the formula attempts to manage the salt-minion or salt-master services.

### Attempts to fix the bug
I have not attempted to fix the bug itself. I am able to prevent the behavior from occurring by switching my ref in my salt-master configuration to db8570609ce3a167a4759f03fec42ef2db28093d.

### PR DESCRIPTION
**Pull request by noelmcloughlin**
### PR progress checklist (to be filled in by reviewers)

- [x] Changes to documentation are appropriate (or tick if not required)
- [x] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [x] `[docs]`     Documentation changes
- [x] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Related issues and/or pull requests

Fix #475 

### Describe the changes you're proposing

See #475 

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

File: salt/minion.sls [hunk 1]
```diff
@@ -104,8 +106,7 @@ salt-minion:
     - enable: {{ salt_settings.minion_service_details.enabled }}
     - name: {{ salt_settings.minion_service }}
     - watch:
-      - file: salt-master-macos
-      - file: salt-master
+      # file: salt-master
```
Comment (ixs) [inline salt/minion.sls:109]: I would not comment it out, I would remove the line if it's not needed.

File: salt/master.sls [hunk 2]
```diff
@@ -55,8 +55,11 @@ salt-master:
     - enable: {{ salt_settings.master_service_details.enabled }}
     - name: {{ salt_settings.master_service }}
     - watch:
+           {%- if grains.kernel|lower == 'darwin' %}
```
Comment (ixs) [inline salt/master.sls:58]: Logic here looks better.

Comment (noelmcloughlin) [reply salt/master.sls:58]: is this okay?

File: salt/minion.sls (hunk 1 above)
Comment (noelmcloughlin) [reply salt/minion.sls:109]: yes, thanks.

**Comment by prannonpendragas** [comment]
Tested:

Gitfs config for this formula below:

```
    - https://github.com/noelmcloughlin/salt-formula.git:
      - all_saltenvs: a46645607b3e20cf2f69a51e976457c201830c4c
```

Executions on master node:

```
[08:34:16][root]@[foundation01][/srv]$ salt-call state.apply salt.minion
local:
----------
          ID: salt-minion
    Function: pkg.installed
      Result: True
     Comment: All specified packages are already installed
     Started: 08:39:03.023091
    Duration: 32.669 ms
     Changes:   
----------
          ID: salt-minion
    Function: file.recurse
        Name: /etc/salt/minion.d
      Result: True
     Comment: The directory /etc/salt/minion.d is in the correct state
     Started: 08:39:03.058575
    Duration: 375.924 ms
     Changes:   
----------
          ID: remove-old-minion-conf-file
    Function: file.absent
        Name: /etc/salt/minion.d/_defaults.conf
      Result: True
     Comment: File /etc/salt/minion.d/_defaults.conf is not present
     Started: 08:39:03.435386
    Duration: 0.285 ms
     Changes:   
----------
          ID: salt-minion
    Function: service.running
      Result: True
     Comment: The service salt-minion is already running
     Started: 08:39:03.435826
    Duration: 42.624 ms
     Changes:   
----------
          ID: salt-minion
    Function: cmd.run
        Name: salt-call --local service.restart salt-minion --out-file /dev/null
      Result: True
     Comment: State was not run because none of the onchanges reqs changed
     Started: 08:39:03.480725
    Duration: 0.008 ms
     Changes:   

Summary for local
------------
Succeeded: 5
Failed:    0
------------
Total states run:     5
Total run time: 451.510 ms
```

```
[08:39:03][root]@[foundation01][/srv]$ salt-call state.apply salt.master
local:
----------
          ID: salt-master
    Function: pkg.installed
      Result: True
     Comment: All specified packages are already installed
     Started: 08:40:04.787669
    Duration: 32.517 ms
     Changes:   
----------
          ID: salt-master
    Function: file.recurse
        Name: /etc/salt/master.d
      Result: True
     Comment: Recursively updated /etc/salt/master.d
     Started: 08:40:04.823066
    Duration: 397.693 ms
     Changes:   
              ----------
              /etc/salt/master.d/f_defaults.conf:
                  ----------
                  diff:
                      --- 
                      +++ 
                      @@ -847,8 +847,8 @@
                             - all_saltenvs: master
                           - https://github.com/compunautics/rundeck-formula.git:
                             - all_saltenvs: master
                      -    - https://github.com/noelmcloughlin/salt-formula.git:
                      -      - all_saltenvs: a46645607b3e20cf2f69a51e976457c201830c4c
                      +    - https://github.com/saltstack-formulas/salt-formula.git:
                      +      - all_saltenvs: db8570609ce3a167a4759f03fec42ef2db28093d
                           - https://github.com/compunautics/sssd-formula.git:
                             - all_saltenvs: master
                           - https://github.com/compunautics/telegraf-formula.git:
              removed:
                  - /etc/salt/master.d/.f_defaults.conf.swp
----------
          ID: remove-old-master-conf-file
    Function: file.absent
        Name: /etc/salt/master.d/_defaults.conf
      Result: True
     Comment: File /etc/salt/master.d/_defaults.conf is not present
     Started: 08:40:05.228020
    Duration: 0.291 ms
     Changes:   
----------
          ID: salt-master
    Function: service.running
      Result: True
     Comment: Service restarted
     Started: 08:40:05.280641
    Duration: 960.448 ms
     Changes:   
              ----------
              salt-master:
                  True

Summary for local
------------
Succeeded: 4 (changed=2)
Failed:    0
------------
Total states run:     4
Total run time:   1.391 s
```

Appears to have removed the previous error.

**Comment by noelmcloughlin** [comment]
Can you formally review the PR @ixs  and we can merge.

**Comment by myii** [comment]
Merged, thanks to all involved for getting this resolved.

**Pull request merged**